debezium-core/src/main/java/io/debezium/relational/history/AbstractFileBasedSchemaHistory.java

```diff
@@ -124,7 +124,19 @@ protected void storeRecord(HistoryRecord record) throws SchemaHistoryException {
 
     @Override
     protected void recoverRecords(Consumer<HistoryRecord> records) {
-        lock.write(() -> getRecords().forEach(records));
+        lock.write(() -> {
+            for (HistoryRecord record : getRecords()) {
+                records.accept(record);
+                try {
+                    LOGGER.info("Sleeping for {} ms to emulate CPU-intensive environment", config.getLong(SCHEMA_HISTORY_RECOVERY_DELAY_MS));
```

Comment:
This class is used for non-test components too so it must not have something like it hard-coded. For testing a new test only schema history implementation should be provided.